server/auth/types/proxy/proxy_auth.ts

```diff
@@ -69,88 +85,70 @@ export class ProxyAuthentication implements IAuthenticationType {
       // TODO: check how to get remoteIp from KibanaRequest and add remoteIp to this header
       authHeaders[customProxyHeader] = this.config.proxycache!.proxy_header_ip;
     }
+    return authHeaders;
+  }
 
-    if (cookie) {
-      if (get(cookie.credentials, this.userHeaderName)) {
-        authHeaders[this.userHeaderName] = cookie.credentials[this.userHeaderName];
-        if (get(cookie.credentials, this.roleHeaderName)) {
-          authHeaders[this.roleHeaderName] = cookie.credentials[this.roleHeaderName];
-        }
-        if (get(cookie.credentials, ProxyAuthentication.XFF)) {
-          authHeaders[ProxyAuthentication.XFF] = cookie.credentials[ProxyAuthentication.XFF];
-        }
-
-        cookie.expiryTime = Date.now() + this.config.cookie.ttl;
-        this.sessionStorageFactory.asScoped(request).set(cookie);
-        return toolkit.authenticated({
-          requestHeaders: authHeaders,
-        });
-      } else if (get(cookie.credentials, 'authorization')) {
-        authHeaders.authorization = get(cookie.credentials, 'authorization');
-        return toolkit.authenticated({
-          requestHeaders: authHeaders,
-        });
-      }
-    }
-
-    // no credentials in the cookie, fall back to do authentication with header
-    if (request.headers[this.userHeaderName]) {
-      authHeaders[this.userHeaderName] = request.headers[this.userHeaderName];
+  getCookie(request: KibanaRequest, authInfo: any): SecuritySessionCookie {
+    const cookie: SecuritySessionCookie = {
+      username: authInfo.username,
+      credentials: {},
+      authType: this.authType,
+      isAnonymousAuth: false,
+      expiryTime: Date.now() + this.config.cookie.ttl,
+    };
+    if (this.userHeaderName && request.headers[this.userHeaderName]) {
+      cookie.credentials[this.userHeaderName] = request.headers[this.userHeaderName];
     }
-    if (this.roleHeaderName && request.headers[this.roleHeaderName]) {
-      authHeaders[this.roleHeaderName] = request.headers[this.roleHeaderName];
+    if (this.roleHeaderName) {
```

Comment:
Just want to confirm that this is the expected behavior. Looks like this is the only one which does not check the existence of value before assigning it.

Reply:
`user` is mandatory, while role is not, so it doesn't matter if role header is set to undefined. updated to check though.